build.gradle.kts

```diff
@@ -0,0 +1,31 @@
+plugins {
+    `embedded-kotlin`
```

Comment:
One last question: Is this `embedded-kotlin` documented anywhere? All I can find is this [entry](https://plugins.gradle.org/plugin/org.gradle.kotlin.embedded-kotlin/2.3.3) on the gradle plugin portal, which links to an archived github repository. I presume this repo contains the precursor to the modern kotlin DSL support.

I am familiar with .kts build scripts and also the `kotlin-dsl` plugin (documented [here](https://docs.gradle.org/current/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin)), but wondering about the plugin applied here.

(Digression: One aspect I find confusing when developing a Gradle plugin in Kotlin is the compatibility pitfalls regarding the kotlin stdlib version this plugin/gradle uses vs. the kotlin stdlib version used by a project that applies this plugin. If you ever get around to make a video about that, I would be interested).

Reply:
Yes the link there is outdated it seems. It all was integrated into https://github.com/gradle/gradle at some point.

> (Digression: One aspect I find confusing when developing a Gradle plugin in Kotlin is the compatibility pitfalls regarding the kotlin stdlib version this plugin/gradle uses vs. the kotlin stdlib version used by a project that applies this plugin. If you ever get around to make a video about that, I would be interested).

Yes I agree and it is the source of much confusion and discussion. Right now I would say it is best, when doing Gradle plugin development in Kotlin, to somehow use the version that is embedded in Gradle. Because that's the only version Gradle can use at runtime.

Looks like `embedded-kotlin` is not documented then. 😐 It is basically `kotlin.jvm` with the version embedded in Gradle. That's why I used it as it fits what you had before minus the version.

`kotlin-dsl` also applies `embedded-kotlin` under the hood and adds more conveniences on top. One thing it adds is a Kotlin compile plugin (SAM-with-receiver) that allows you to use `this` instead of `it` in closures. Which makes  the DSL more compact (and more similar to the Groovy DSL variant). We could also use that, but we might need to adjust the code in a few places then.